### PR TITLE
Proj0032: Migarate away from BinaryFormatter

### LINKED ISSUE #253: New rule:  EnableUnsafeBinaryFormatterSerialization is strongly recommended against
`BinaryFormatter` has been obsolete for a [while](https://learn.microsoft.com/en-us/dotnet/core/compatibility/serialization/5.0/binaryformatter-serialization-obsolete), but still usable through `<EnableUnsafeBinaryFormatterSerialization>true</EnableUnsafeBinaryFormatterSerialization>`. In .NET 9 `BinaryFormatter` has been [removed](https://devblogs.microsoft.com/dotnet/binaryformatter-removed-from-dotnet-9/) and it is strongly recommended to not use it (through the unsupported `System.Runtime.Serialization.Formatters` package) and `EnableUnsafeBinaryFormatterSerialization` anymore.

### PR DESCRIPTION
**Pull request by Corniel**
Closes #253

### CODE REVIEW

File: specs/DotNetProjectFile.Analyzers.Specs/Rules/MS_Build/Migrate_away_from_BinaryFormatter.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
+namespace Rules.MS_Build.Migrate_away_from_BinaryFormatter;
+
+public class Reports
+{
+    [Test]
+    public void when_BinaryFormatter_is_enabled() => new MigrateAwayFromBinaryFormatter()
+        .ForProject("BinaryFormattingEnabled.cs")
+        .HasIssue(Issue.WRN("Proj0032", "Migrate away from BinaryFormatter and disable <EnableUnsafeBinaryFormatterSerialization>.")
```
Comment (CptWesley) [inline specs/DotNetProjectFile.Analyzers.Specs/Rules/MS_Build/Migrate_away_from_BinaryFormatter.cs:8]: The .csproj under tests initially enables, then later disables this flag, so one could argue it shouldn't report this options.

File: src/DotNetProjectFile.Analyzers/Rule.cs [hunk 2]
```diff
@@ -284,6 +284,16 @@ public static partial class Rule
         tags: ["clarity", "casing"],
         category: Category.Clarity);
 
+    public static DiagnosticDescriptor MigrateAwayFromBinaryFormatter => New(
+        id: 0032,
+        title: "Migrate away from BinaryFormatter",
+        message: "Migrate away from BinaryFormatter and disable <{0}>.",
```
Comment (CptWesley) [inline src/DotNetProjectFile.Analyzers/Rule.cs:290]: I'd argue that removing the line entirely should be preferred over disabling it, but perhaps that should be its own rule

File: specs/DotNetProjectFile.Analyzers.Specs/Rules/MS_Build/Migrate_away_from_BinaryFormatter.cs (hunk 1 above)
Comment (Corniel) [reply specs/DotNetProjectFile.Analyzers.Specs/Rules/MS_Build/Migrate_away_from_BinaryFormatter.cs:8]: There is no reason to enable it in the first place, so I'll report on setting it to true. I've even considered to report on setting it to false too due to that reasoning.

File: src/DotNetProjectFile.Analyzers/Rule.cs (hunk 2 above)
Comment (Corniel) [reply src/DotNetProjectFile.Analyzers/Rule.cs]: You're right. I've rephrased the message.

**Pull request merged**